### PR TITLE
some mathematical algorithm added

### PR DESCRIPTION
**Pull request by axshivam**

### CODE REVIEW

**Comment by Bhavnaharitsa** [comment]
Please mention the name of algorithm in pr title

File: Algorithms/towerofhanoi.c [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-void main()
-{
-    TowerOfHanoi(3,'s','m','d');
-}
+#include<stdio.h>
+void TowerOfHanoi(int size, char c1, char c2, char c3)
+{
+    if(size==1)
+    {
+        printf("%d  Disc  transferred from %c to %c \n", size, c1, c3);
```
Comment (Bhavnaharitsa) [inline Algorithms/towerofhanoi.c]: What's the difference between previous one and this one?

**Comment by Bhavnaharitsa** [comment]
Too many algorithms under a single commit
Proper way is to.open a directory and then execute one by one

**Comment by axshivam** [comment]
> Please mention the name of algorithm in pr title

done

File: Algorithms/towerofhanoi.c (hunk 1 above)
Comment (axshivam) [reply Algorithms/towerofhanoi.c]: > Too many algorithms under a single commit
> Proper way is to.open a directory and then execute one by one

```Sorry for all the mistakes , i am beginner in this field```

**Comment by Bhavnaharitsa** [comment]
> Too many algorithms under a single commit
> Proper way is to.open a directory and then execute one by 

 there are too many algorithms for one pr.

**Comment by axshivam** [comment]
> > Too many algorithms under a single commit
> > Proper way is to.open a directory and then execute one by
> 
> there are too many algorithms for one pr.

so what will i do? plz guide me

File: Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			for(int j=i*i; j<=n;j+=i)
+			{
+				prime[j]=false;
+			}
+		}
+	}
+	for(int p=2;p<=n;p++)
+	{
+		if(prime[p])
+		cout<<p<<" ";
```
Comment (Bhavnaharitsa) [inline Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp:27]: why this unwanted quotations

File: Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	}
+	cout<<endl;
+}
+int main()
+{
+	ios_base::sync_with_stdio(false);
+	cin.tie(NULL); cout.tie(NULL);
+	int n;
```
Comment (Bhavnaharitsa) [inline Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp]: wrong declaration of variable name

File: Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Given a number n , print all primes number smaller than or equal to n.
+// it is algo given n is smaller number
+#include<bits/stdc++.h>
+using namespace std;
+void Sieve(int n)
```
Comment (Bhavnaharitsa) [inline Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp]: wrong declaration of variable

**Comment by Bhavnaharitsa** [comment]
change all single letter variable names in all algorithms and give proper initialisation, unwanted quotations, open a directory, shove all the algorithms.

Remember next time,  do not make such changes or add too many algorithms in one pr. 
Every pr is supposed to have one algorithm with a relevant issue open. This is violating the rules

**Comment by axshivam** [comment]
> change all single letter variable names in all algorithms and give proper initialisation, unwanted quotations, open a directory, shove all the algorithms.
> 
> Remember next time, do not make such changes or add too many algorithms in one pr.
> Every pr is supposed to have one algorithm with a relevant issue open. This is violating the rules

Thanks for explaning

File: Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp (hunk 2 above)
Comment (axshivam) [reply Algorithms/mathematical_algorithm/Sieve_of_Eratosthenes.cpp:27]: without quotations ouput look like this
```Enter the number:```
```89```
```23571113171923293137414347535961677173798389```

**Comment by Bhavnaharitsa** [comment]
Resolve conflicts and we are good to go

**Comment by axshivam** [comment]
conflicts resolved

**Comment by Bhavnaharitsa** [comment]
I can merge this but if you have been contributing for NWoC the period has ended. And hence, it will not lead to any points.

**Comment by axshivam** [comment]
Can i work more on this??

**Comment by Bhavnaharitsa** [comment]
Which one?

**Comment by axshivam** [comment]
I mean can i add more algorithm on this repo??

**Comment by Bhavnaharitsa** [comment]
yes, you can

**Pull request merged**